urllib3/exceptions.py

```diff
@@ -112,6 +112,9 @@ class ConnectTimeoutError(TimeoutError):
     "Raised when a socket timeout occurs while connecting to a server"
     pass
 
+class NewConnectionError(ConnectTimeoutError, PoolError):
```

Comment:
The error happens in the pool though, when `._new_conn()` is called. Also `ProtocolError` is supposed to be for errors that happen mid-request, as opposed to strictly before the request is even sent.